runtime/compiler/runtime/StatisticsThread.cpp

```diff
@@ -0,0 +1,172 @@
+   }
+
+static int32_t J9THREAD_PROC statisticsThreadProc(void * entryarg)
+   {
+   J9JITConfig * jitConfig = (J9JITConfig *) entryarg;
+   J9JavaVM * vm = jitConfig->javaVM;
+
+   TR_StatisticsThread *statisticsThread = ((TR_JitPrivateConfig*)(jitConfig->privateConfig))->statisticsThread;
+   UDATA samplingPeriod    = std::max(static_cast<UDATA>(TR::Options::_minSamplingPeriod), jitConfig->samplingFrequency);
+
+   J9VMThread *statThread = NULL;
+   PORT_ACCESS_FROM_JAVAVM(vm);
+
+   int rc = vm->internalVMFunctions->internalAttachCurrentThread(vm, &statThread, NULL,
+                                  J9_PRIVATE_FLAGS_DAEMON_THREAD | J9_PRIVATE_FLAGS_NO_OBJECT |
+                                  J9_PRIVATE_FLAGS_SYSTEM_THREAD | J9_PRIVATE_FLAGS_ATTACHED_THREAD,
+                                  statisticsThread->getStatisticsOSThread());
+   statisticsThread->getStatisticsThreadMonitor()->enter();
+   statisticsThread->setAttachAttempted(true);
+   if (rc != JNI_OK)
+      return JNI_ERR; // attaching the JITaaS Statistics thread failed
+
+   statisticsThread->setStatisticsThread(statThread);
+   statisticsThread->getStatisticsThreadMonitor()->notifyAll();
+   statisticsThread->getStatisticsThreadMonitor()->exit();
+
+   j9thread_set_name(j9thread_self(), "JITaaS Server Statistics Thread");
+
+   TR::CompilationInfo *compInfo = TR::CompilationInfo::get(jitConfig);
+   TR::PersistentInfo *persistentInfo = compInfo->getPersistentInfo();
+   uint64_t crtTime = j9time_current_time_millis();
+   uint64_t oldSyncTime = crtTime;
+   uint64_t lastStatistics = crtTime;
+
+   persistentInfo->setStartTime(crtTime);
+   persistentInfo->setElapsedTime(0);
+   while(!statisticsThread->getStatisticsThreadExitFlag())
+      {
+      while(!statisticsThread->getStatisticsThreadExitFlag() && j9thread_sleep_interruptable((IDATA) samplingPeriod, 0) == 0)
+         {
+         crtTime = j9time_current_time_millis();
+         persistentInfo->setElapsedTime(crtTime - persistentInfo->getStartTime());
+
+         if ((statisticsThread->getStatisticsFrequency()) && ((crtTime - lastStatistics) > statisticsThread->getStatisticsFrequency()))
+            {
+         //   TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "No of clients : 0x%x \n", ClientSessionHT::_clientSessionMap->size());
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Total thread : %d \n", compInfo->getNumUsableCompilationThreads());
+            TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "Active Thread : %d \n",compInfo->getNumCompThreadsActive());
+            CpuUtilization *cpuUtil = compInfo->getCpuUtil();
+            if (cpuUtil->isFunctional())
+               {
+
+               TR_VerboseLog::writeLineLocked(TR_Vlog_JITaaS, "CpuLoad %d%% (AvgUsage %d%%) JvmCpu %d%% \n", cpuUtil->getCpuUsage(), cpuUtil->getAvgCpuUsage(), cpuUtil->getVmCpuUsage());
+               }
+            lastStatistics = crtTime;
+            }
+            oldSyncTime = crtTime;
```

Comment:
`oldSyncTime` is not needed anymore because you read the correct time every single time